src/ts/services/TelemetryService.ts

```diff
@@ -1,69 +1,55 @@
-import {
-  ITyperighterTelemetryEvent,
-  ISuggestionAcceptedEvent,
-  TYPERIGHTER_TELEMETRY_TYPE,
-  IMarkAsCorrectEvent,
-  ISidebarClickEvent
-} from "../interfaces/ITelemetryData";
+import throttle from 'lodash/throttle';
+import chunk from 'lodash/chunk';
 
-class TelemetryService {
-  constructor(
-    private sendTelemetryEvent: (event: ITyperighterTelemetryEvent) => void
-  ) {}
+import { ITyperighterTelemetryEvent } from "../interfaces/ITelemetryData";
 
-  public suggestionIsAccepted(tags: ISuggestionAcceptedEvent["tags"]) {
-    this.sendTelemetryEvent({
-      type: TYPERIGHTER_TELEMETRY_TYPE.TYPERIGHTER_SUGGESTION_IS_ACCEPTED,
-      value: 1,
-      eventTime: new Date().toISOString(),
-      tags
-    });
-  }
+class UserTelemetryEventSender {
+    private postEventLimit = 500;
+    private eventBuffer: ITyperighterTelemetryEvent[] = [];
 
-  public matchIsMarkedAsCorrect(tags: IMarkAsCorrectEvent["tags"]) {
-    this.sendTelemetryEvent({
-      type: TYPERIGHTER_TELEMETRY_TYPE.TYPERIGHTER_MARK_AS_CORRECT,
-      value: 1,
-      eventTime: new Date().toISOString(),
-      tags
-    });
-  }
+    public constructor(private telemetryUrl: string, private throttleDelay: number = 10000) {};
 
-  public documentIsChecked(tags: ITyperighterTelemetryEvent["tags"]) {
-    this.sendTelemetryEvent({
-      type: TYPERIGHTER_TELEMETRY_TYPE.TYPERIGHTER_CHECK_DOCUMENT,
-      value: 1,
-      eventTime: new Date().toISOString(),
-      tags
-    });
-  }
+    private async sendEvents(): Promise<void> {
+        const [firstChunk, ...subsequentChunks] = chunk(this.eventBuffer, this.postEventLimit);
+        if (!firstChunk) {
+          return Promise.resolve();
+        }
 
-  public typerighterIsOpened(tags: ITyperighterTelemetryEvent["tags"]) {
-    this.sendTelemetryEvent({
-        type: TYPERIGHTER_TELEMETRY_TYPE.TYPERIGHTER_OPEN_STATE_CHANGED,
-        value: true,
-        eventTime: new Date().toISOString(),
-        tags
-    });
-  }
+        const jsonEventBuffer = JSON.stringify(firstChunk);
 
-  public typerighterIsClosed(tags: ITyperighterTelemetryEvent["tags"]) {
-    this.sendTelemetryEvent({
-        type: TYPERIGHTER_TELEMETRY_TYPE.TYPERIGHTER_OPEN_STATE_CHANGED,
-        value: false,
-        eventTime: new Date().toISOString(),
-        tags
-    });
-  }
+        // Push the remaining events back into the buffer
+        this.eventBuffer = subsequentChunks.flat();
 
-  public sidebarMatchClicked(tags: ISidebarClickEvent["tags"]) {
-    this.sendTelemetryEvent({
-        type: TYPERIGHTER_TELEMETRY_TYPE.TYPERIGHTER_SIDEBAR_MATCH_CLICK,
-        value: 1,
-        eventTime: new Date().toISOString(),
-        tags
-    });
-  }
+        const response = await fetch(`${this.telemetryUrl}/event`, {
```

Comment:
Would it be better not to assume a certain endpoint is on the url and trust the the parameter parsed in is correct?

Reply:
This anticipates being integrated in the telemetry service, where it's safe to have that opinion! It's a bit strange now, but it shouldn't be here for long.